ddtrace/monkey.py

```diff
@@ -56,127 +46,96 @@
+    :type integration: str
+    :param raise_errors: whether or not to raise errors if they occur
+    :type raise_errors: bool
     """
-    modules = PATCH_MODULES.copy()
-    modules.update(patch_modules)
+    intmodname = integration_modname(integration)
 
-    patch(raise_errors=False, **modules)
+    try:
+        intmod = importlib.import_module(intmodname)
+    except ImportError:
+        log_msg = 'install: integration "{}" not found'.format(integration)
+        if raise_errors:
+            raise PatchException(log_msg)
+        else:
+            log.error(log_msg)
+            return
 
+    if not hasattr(intmod, 'patch'):
+        log.error('install: integration "{}" does not have patch attribute'.format(integration))
+        return
 
-def patch(raise_errors=True, **patch_modules):
-    """Patch only a set of given modules.
+    intmod.patch()
 
-    :param bool raise_errors: Raise error if one patch fail.
-    :param dict \**patch_modules: List of modules to patch.
 
-        >>> patch(psycopg=True, elasticsearch=True)
-    """
-    modules = [m for (m, should_patch) in patch_modules.items() if should_patch]
-    for module in modules:
-        if module in _PATCH_ON_IMPORT:
-            # If the module has already been imported then patch immediately
-            if module in sys.modules:
-                patch_module(module, raise_errors=raise_errors)
-
-            # Otherwise, add a hook to patch when it is imported for the first time
-            else:
-                # Use factory to create handler to close over `module` and `raise_errors` values from this loop
-                when_imported(module)(_on_import_factory(module, raise_errors))
-
-                # manually add module to patched modules
-                _PATCHED_MODULES.add(module)
-        else:
-            patch_module(module, raise_errors=raise_errors)
+def patch_all(**integrations):
+    """Patch all default enabled integrations.
 
-    patched_modules = get_patched_modules()
-    log.info(
-        "patched %s/%s modules (%s)",
-        len(patched_modules),
-        len(modules),
-        ",".join(patched_modules),
-    )
+    :param **integrations: override default enabled integrations.
+    :type integrations: dict
 
+    To patch all the default enabled integrations::
 
-def patch_module(module, raise_errors=True):
-    """Patch a single module
+        patch_all()
 
-    Returns if the module got properly patched.
+    To patch all the default enabled integrations, but  specific ones::
+
+        patch_all(redis=False, futures=True)
     """
-    try:
-        return _patch_module(module)
-    except Exception as exc:
-        if raise_errors:
-            raise
-        log.debug("failed to patch %s: %s", module, exc)
-        return False
+    install_all(overrides=integrations, raise_errors=False)
+
 
+def patch(raise_errors=True, **integrations):
+    """Patch specified integrations.
 
-def get_patched_modules():
```

Comment:
Why are we removing this?

Reply:
This function was used for the logging to display the modules that were found (imported) and patched. To me, it no longer makes sense to have this since we will be "patching" (installing the import hooks) for all modules. This method originally returned the list of **integrations** that were used. With the new changes, would this method mean get all modules-to-be-patched that have import hooks installed or would it be the integrations that we have installed import hooks for or maybe even modules-to-be-patched in which the hooks have run for? 

We may want to think about our terminology within the project when referring to

- what we call an integration
- the name of python module that one of our integration(s) deal with
- when something is "patched"
    - does this mean import hooks installed or does it mean that the library has actually been imported and our patching has been done on it
- enabling and disabling integrations
- enabling and disabling the installation of integrations

all of these are new or changed things within the library that we should consider defining.